### PR TITLE
Messaging

### PR DESCRIPTION
**Pull request by raman-nbg**
See issue: https://github.com/soumak77/firebase-mock/issues/164

### CODE REVIEW

**Comment by raman-nbg** [comment]
@dmurvihill had you already time to have a look on this?

File: test/unit/messaging.js [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+      messaging.flush();
+      thenable.then(function() {
+        done('send() should not resolve');
+      }).catch(function() {
+        done();
+      });
+    });
+
+    it('can return user defined results', function(done) {
+      messaging.nextResult('send', 'the result');
```
[the remark this answers is not in the document]
Comment (raman-nbg) [reply test/unit/messaging.js]: Sounds good

File: tutorials/admin/messaging.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+MockFirebase replaces most of Firebase's messaging method of the admin API with simple mocks. Messaging methods will always succeed unless an error is specifically specified using [`failNext`](../API.md#failnextmethod-err---undefined).
+
+## Send message
+
+In this example, we'll create send a new message using Firebase messaging.
+
+##### Source
+
+```js
+var ref;
```
[the remark this answers is not in the document]
Comment (raman-nbg) [reply tutorials/admin/messaging.md]: Added

File: tutorials/admin/messaging.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
+# Tutorial: Authentication
```
[the remark this answers is not in the document]
Comment (raman-nbg) [reply tutorials/admin/messaging.md]: Fixed

**Comment by raman-nbg** [comment]
Fixed everything you mentioned :)

File: API.md [hunk 4]
```diff
@@ -19,6 +19,10 @@ Only `MockFirebase` methods are included here. For details on normal Firebase AP
 - [Server Timestamps](#server-timestamps)
   - [`setClock(fn)`](#firebasesetclockfn---undefined)
   - [`restoreClock()`](#firebasesetclockfn---undefined)
+- [Messaging](#messaging)
+  - [`respondNext(methodName, result)`](#respondnextmethodname--result---undefined)
```
[the remark this answers is not in the document]
Comment (raman-nbg) [reply API.md]: Fixed.

File: tutorials/admin/messaging.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
+# Tutorial: Messaging
+
+MockFirebase replaces most of Firebase's messaging method of the admin API with simple mocks. Messaging methods will always succeed unless an error is specifically specified using [`failNext`](../API.md#failnextmethod-err---undefined).
```
[the remark this answers is not in the document]
Comment (raman-nbg) [reply tutorials/admin/messaging.md]: Fixed.

**Comment by dmurvihill** [comment]
Merged. Thanks!

**Pull request merged**